File: commands/photo.py
#!/usr/bin/env python3
import os
import boto3
import random
import discord


async def photo(ctx):

    embed = discord.Embed(color=0xFEE9B6, title="⏳  Posting Random Photo...")
    interaction = await ctx.respond(embed=embed)

    TABLE = "discord-photos"
    ddb = boto3.resource("dynamodb")
    table = ddb.Table(TABLE)

    data = table.scan()["Items"]
    name = random.choice(data)["name"]

    url = boto3.client("s3").generate_presigned_url(
        ClientMethod="get_object",
        Params={"Bucket": str(os.getenv("PHOTO_BUCKET")), "Key": name},
        ExpiresIn=604800,
    )

    embed = discord.Embed(color=0xFFCC4D, title="📸  Random Photo").set_image(url=url)
    await interaction.edit_original_response(embed=embed)

File: commands/photo.ts
import {
  DiscordInteraction,
  InteractionResponse,
  Colors,
} from "../types/discord";
import { embedResponse, errorResponse } from "../utils/discord";
import { getRandomPhoto, getPhotoUrl } from "../services/photos";

export async function handlePhoto(
  _interaction: DiscordInteraction,
): Promise<InteractionResponse> {
  try {
    const photo = await getRandomPhoto();

    if (!photo) {
      return errorResponse("No photos in the database yet!");
    }

    const url = await getPhotoUrl(photo.name);

    return embedResponse({
      title: "📸 Random Photo",
      color: Colors.YELLOW,
      image: { url },
    });
  } catch (error) {
    console.error("Photo error:", error);
    return errorResponse("Failed to fetch a random photo.");
  }
}
